src/pal/src/exception/signal.cpp

```diff
@@ -402,11 +409,23 @@ Return :
 --*/
 bool IsRunningOnAlternateStack(void *context)
 {
-    stack_t *signalStack = &((native_context_t *)context)->uc_stack;
-    // Check if the signalStack local variable address is within the alternate stack range. If it is not,
-    // then either the alternate stack was not installed at all or the current method is not running on it.
-    void* alternateStackEnd = (char *)signalStack->ss_sp + signalStack->ss_size;
-    return ((signalStack->ss_flags & SS_DISABLE) == 0) && (signalStack->ss_sp <= &signalStack) && (&signalStack < alternateStackEnd);
+    bool isRunningOnAlternateStack;
+    if (g_enable_alternate_stack_check)
```

Comment:
Nit: is it worth adding a comment describing that not all Linux environments allow/support detection of alt stack in this way?

Reply:
Sounds reasonable. Actually, only the WSL doesn't support it (and I expect that WSL2 will support it).